js/battle.js

```diff
@@ -6147,6 +6155,14 @@ var Battle = (function () {
 					if (slot >= 0) pokemon.slot = slot;
 					return pokemon;
 				}
+				if (!pokemon.searchid && !pokemon.shiny && pokemon.checkDetails(details.replace(', shiny', ''))) { // switch-in matches non-shiny Team Preview entry
```

Comment:
You should possibly drag the entire `switch-in matches Team Preview entry` case down here, instead of leaving the cases messy.

Reply:
From what I've been able to tell, `pokemon.needsReplace` is used for the Arceus/Silvally/etc. that shouldn't reveal their typing on Team Preview. If I move that down, is there any remaining case where

    if (pokemon.needsReplace) {
        pokemon = this.p2.newPokemon(this.parseDetails(name, pokemonid, details), i);
    }

happens?

Reply:
Probably not.

Reply:
I would recommend removing it, yes.